### DISCUSSION TITLE
Equilizer3Bands usage

### DISCUSSION THREAD

**Discussion by geoffw64** [pschatzmann/arduino-audio-tools#1604] [Q&A] [answered]
Hi Phil, 
I see you've added a 3 band equaliser to the library.  Not sure if you want help with testing but I decided to give it a go.  
Is this the right technique to use the equaliser with volume control? 

  VolumeStream volume_stream(I2S_stream);  
  Equilizer3Bands equilized_stream(volume_stream); 
  ConfigEquilizer3Bands eq_config;
 ... followed by.. 
  volume_stream.setVolume(0.1);  
  eq_config.gain_low = 1.0; 
  eq_config.gain_medium = 0.7; 
  eq_config.gain_high = 0.7;  
  dec.setOutput(equilized_stream); // Sends stream with reduced volume & equilisation
...

The equalisation seems to work nicely, however I am getting louder volume than expected.  Have I used it incorrectly? 
I'm assuming that a gain value of 1.0 means no attenuation or boost, although i haven't traced through the code in detail.

**Comment by pschatzmann** [comment, marked as answer]
Most likly you forgot to call begin on the VolumeStream

**Reply by geoffw64** [reply]
Yes, that was it.  :) Thanks.

**Comment by jumperrg** [comment]
Hi, I'm also trying to implement it, but I didn't find any examples. Did you removed it?

**Comment by pschatzmann** [comment]
There is nothing special about this and you would just build an processing chain like described in Introduction Wiki: Changing the Signal!

Here is the [example using it with A2DP](https://github.com/pschatzmann/arduino-audio-tools/blob/2dd77429dd81a4421ee960ff9a241ffec9e5e977/examples/examples-communication/a2dp/basic-a2dp-eq-audiokit/basic-a2dp-eq-audiokit.ino#L16).
